src/test/ui/linkers/gcc_ld_mismatch.rs

```diff
@@ -0,0 +1,7 @@
+// build-fail
+// compile-flags: -Zgcc-ld=lld -Clinker-flavor=em -Zunstable-options
+
+// Test ensuring that until the unstable flag is removed (if ever), if both the linker-flavor and
+// `gcc-ld` flags are used, they ask for the same linker.
```

Comment:
If making these two options incompatible helps to remove at least some code or conditions in `fn handle_cli_linker_flavors` then please let's do that instead.